orbit/pkg/profiles/profiles_darwin.go

```diff
@@ -104,3 +106,121 @@ var getMDMInfoFromProfilesCmd = func() ([]byte, error) {
 	cmd := exec.Command("/usr/bin/profiles", "status", "-type", "enrollment")
 	return cmd.Output()
 }
+
+// CheckAssignedEnrollmentProfile runs the `profiles show -type enrollment` command to get the assigned
+// MDM enrollment profile and reports if the hostname of the MDM server
+// in the assigned profile the device matches the hostname of the provided URL.
+func CheckAssignedEnrollmentProfile(expectedURL string) error {
+	expected, err := url.Parse(expectedURL)
+	if err != nil {
+		return fmt.Errorf("parsing expected URL: %w", err)
+	}
+
+	out, err := showEnrollmentProfileCmd()
+	if err != nil {
+		if exitErr, ok := err.(*exec.ExitError); ok {
+			return fmt.Errorf("show enrollment profile command: %w: %s", err, exitErr.Stderr)
+		}
+		return fmt.Errorf("show enrollment profile command: %w", err)
+	}
+
+	// If an enrollment profile is assigned, the output of the command is in the form:
+	//
+	// ```
+	// Device Enrollment configuration:
+	// {
+	//     AllowPairing = 1;
+	//     AutoAdvanceSetup = 0;
+	//     AwaitDeviceConfigured = 0;
+	//     ConfigurationURL = "https://test.example.com/mdm/apple/enroll?token=1234";
+	//     ConfigurationWebURL = "https://test.example.com/mdm/apple/enroll?token=1234";
+	//     ...
+	// }
+	// ```
+	//
+	// If the host is not enrolled into an MDM, the output of the command is in the form:
+	//
+	// ```
+	// Device Enrollment configuration:
+	// (null)
+	// ```
+	//
+	// We will check that the output is at least 2 lines and contains the expected URL
```

Comment:
we should sacrifice a device and verify if this also works once we get rate limited, I pasted the output of the manpage below, but this is the important bit I'm concerned about:

> Should this occur, a note will be displayed with the enrollment information.

Reply:
It should work because any cached info or whatever "note" is displayed in the case of rate limiting wouldn't match the expected URL and this function will return with an error at line 183, which in turn results in an early return at `/notification` line 104.